src/install.ts

```diff
@@ -0,0 +1,200 @@
+import * as vscode from 'vscode';
+import {outputChannel} from './kclStatus';
+import * as os from 'os';
+import * as path from 'path';
+import axios from 'axios';
+import * as fs from 'fs';
+import { Readable } from 'node:stream';
+
+const KCL_PATH = path.join(os.homedir(), '.kcl');
+const KPM_PATH = path.join(KCL_PATH, 'kpm');
+const KPM_BIN_PARTH = path.join(KPM_PATH, 'bin');
+const GIT_ORG = 'kcl-lang';
+const KCL_REPO = 'kcl';
+const RELEASE_BASE_URL = `https://github.com/${GIT_ORG}/${KCL_REPO}/releases/download`;
+export const KCL_LANGUAGE_SERVER = 'kcl-language-server';
+
+export async function promptInstallLanguageServer(): Promise<string | undefined> {
+	const installOptions = ['Install', 'Cancel'];
+	const selected = await vscode.window.showErrorMessage(
+		`The kcl-language-server is required for KCL code intelliSense. Install now?`,
+		...installOptions
+	);
+	switch(selected) {
+		case 'Install':
+			return installLanguageServer();
+		default:
+			return;
+	}
+}
+
+
+export async function installLanguageServer(): Promise<string | undefined> {
```

Comment:
It is best to add CI unit tests for different platforms here to ensure successful download.

Reply:
added unit test and e2e tests.